Som dev

Storch22 opened the pull request.

Storch22 committed:
first SOM producing some output

Storch22 committed:
add class Args

improving run_nextsomcore.ipynb

Storch22 committed:
TIFF as input

Input Data kann nun auch TIFF File sein - code ist noch nicht fehlerfrei durchgelaufen.

Storch22 committed:
argsPlot.py and plot.py

Arguments for drawing plots from self organizing maps

plot.py file is just created, not  filled with functionality

Storch22 committed:
Zwischenstand - nicht fehlerfrei

plot_som_results0.py läuft erfolgreich durch und generiert .png files. Diese müssen noch in einem Notebook integriert werden, sowie auch die plotfunktionen an sich.

Storch22 committed:
update - plot geospace funktioniert noch nicht

Storch22 committed:
kleine Anpassungen Pfade

Storch22 committed:
plot geospace ok

skript plot_som_results0.py gibt nun auch einen Plot im geospace aus (lokal abgespeichert)

Storch22 committed:
some changes

Use Tiff files do not work yet

Storch22 committed:
reorder file system

Storch22 committed:
changes to jupyter notebook regarding plotting of results

Storch22 committed:
fixed plotting script, file structure reordered

Storch22 committed:
Merge branch 'main' of https://github.com/DARPA-CRITICALMAAS/beak-ta3 into som_dev

Storch22 committed:
use GeoTIF as input to SOM

Changes to load_geotiff - delete data rows with nan, same for X Y coordinates

Storch22 committed:
noData handling

when input data is GeoTiff, noDataValue is set to np.nan once data is read as array. This allows to have input data with different data type (int, float).

In KMeans clustering the parameter n_init is explicitely set to the default value of 10 from older sklearn versions to avoid warnings. Could also be set to 'auto' instead of 10.

Storch22 committed:
Merge branch 'main' of https://github.com/DARPA-CRITICALMAAS/beak-ta3 into som_dev

Storch22 committed:
Davies Boulduin Index Plot added

Storch22 committed:
minor changes

Commit:
minor: .tif and print status

Storch22 committed:
record time and correct write GeoTIFF

- run SOM: record time and print more progress report;
- write GeoTIFF: correct file name ending (.tif) and correct for loop where strings are selected from the som txt file
-

msmiyels reviewed (commented):
No merging this way. > 3.000 files; we'll do it manually... 🙅

Storch22 committed:
corrections in write_geotiff_out and move_to_subfolder.py

After eliminating nan values, size of data frame changed. Changes in write_geotiff_out were applied to generate X and Y values according to input geoTIF.

In jupyter notebook functionality according to moving files/figures were moved to a separate script: move_to_subfolder.py

Storch22 committed:
write_geotiff_out faster

in read_geotiff_coordinate_columns: return deleted coordinates (nan data)
in save_geospace_result: join coordinates with deleted_coordinates
=> no need to generate new raster matching input raster and the for loops to fill data into that raster

Storch22 committed:
improve efficiency and memory usage in writing results geo / geoTIF

Storch22 committed:
minor clean up after performance improvements

Storch22 committed:
minor changes to ipynb

Storch22 committed:
plot_som_results: bugfix, implove performance

Commit:
plot_som_results: minor changes print()
